flake.nix

```diff
@@ -114,8 +114,49 @@
           flox-cli-tests = callPackage ./pkgs/flox-cli-tests { };
         };
 
+      overlays.development = final: prev: {
+        floxDevelopmentPackages = prev.lib.makeScope prev.newScope (self: {
```

Comment:
Question: can you explain the usage of scopes? From searching around it looks like it creates a new collection of packages to draw from in addition to nixpkgs in a way that the packages in the scope can use each other as inputs?

Reply:
scopes are the nixpkgs intended way to create package groups.

For our case, it mainly provides `callPackage` which you can use to evaluate pacakge expressions with the packages of the scope available (shadowing the same packages in the parent).
We use this below [1] to create the devshell with only the development versions of the packages in scope. the alternative would have been define these as a "normal" attribute set and pulling the pacakges out one by one in the devshell expression.

```nix
# shells/default.nix
{
    flox-cli,
    flox-buildenv,
    flox-package-builder,
    [...]
}: mkShell { [...] }
```

compared to 

```nix
# shells/default.nix
{
   floxDevelopmentPackages
    [...]
}: let 
inherit (floxDevelopmentPackages) 
    flox-cli 
    flox-buildenv
    flox-package-builder
    [...];
in mkShell { [...] } 
```

The latter is still possible with or without scopes if we preferred that.